instrumentation/jsf/jsf-testing-common/src/main/groovy/BaseJsfTest.groovy

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.api.trace.Span.Kind.INTERNAL
+import static io.opentelemetry.instrumentation.test.utils.TraceUtils.basicSpan
+
+import io.opentelemetry.instrumentation.test.AgentTestRunner
+import io.opentelemetry.instrumentation.test.asserts.TraceAssert
+import io.opentelemetry.instrumentation.test.base.HttpServerTestTrait
+import io.opentelemetry.sdk.trace.data.SpanData
+import okhttp3.FormBody
+import okhttp3.HttpUrl
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.Response
+import org.eclipse.jetty.annotations.AnnotationConfiguration
+import org.eclipse.jetty.server.Server
+import org.eclipse.jetty.util.resource.Resource
+import org.eclipse.jetty.webapp.WebAppContext
+import org.jsoup.Jsoup
+
+abstract class BaseJsfTest extends AgentTestRunner implements HttpServerTestTrait<Server> {
+
+  @Override
+  Server startServer(int port) {
+    List<String> configurationClasses = new ArrayList<>()
+    Collections.addAll(configurationClasses, WebAppContext.getDefaultConfigurationClasses())
+    configurationClasses.add(AnnotationConfiguration.getName())
+
+    WebAppContext webAppContext = new WebAppContext()
+    webAppContext.setContextPath(getContextPath())
+    webAppContext.setConfigurationClasses(configurationClasses)
+    // set up test application
+    webAppContext.setBaseResource(Resource.newSystemResource("test-app"))
```

Comment:
Instead of keeping separate app modules for 1.2 and 2.2, would putting them in different directories and changing the resource path work? That way the class extending `BaseJsfTest` would specify by an abstract method which version should be used

Reply:
Done, merged both tests apps into common module.